### ISSUE TITLE
Exception when TLS handshake fails

### ISSUE THREAD

**Issue by wsanchez** [burningmantech/ranger-deploy#41]
From a [build log](https://travis-ci.com/burningmantech/ranger-clubhouse-web/builds/111086950):

```console
Traceback (most recent call last):
  File "/opt/python/3.6.3/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/opt/python/3.6.3/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/deploy/aws/ecs.py", line 709, in <module>
    ECSServiceClient.main()
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/deploy/aws/ecs.py", line 140, in main
    main()
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/deploy/aws/ecs.py", line 575, in staging
    trial_run=trial_run,
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/deploy/notify/smtp.py", line 341, in _staging
    trialRun=trial_run,
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/deploy/notify/smtp.py", line 136, in notifyStaging
    self.smtpHost, self.smtpPort, context=context
  File "/opt/python/3.6.3/lib/python3.6/smtplib.py", line 1030, in __init__
    source_address)
  File "/opt/python/3.6.3/lib/python3.6/smtplib.py", line 251, in __init__
    (code, msg) = self.connect(host, port)
  File "/opt/python/3.6.3/lib/python3.6/smtplib.py", line 336, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/opt/python/3.6.3/lib/python3.6/smtplib.py", line 1038, in _get_socket
    server_hostname=self._host)
  File "/opt/python/3.6.3/lib/python3.6/ssl.py", line 407, in wrap_socket
    _context=self, _session=session)
  File "/opt/python/3.6.3/lib/python3.6/ssl.py", line 814, in __init__
    self.do_handshake()
  File "/opt/python/3.6.3/lib/python3.6/ssl.py", line 1068, in do_handshake
    self._sslobj.do_handshake()
  File "/opt/python/3.6.3/lib/python3.6/ssl.py", line 689, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:777)
```

This should be caught and produce a sane error message.

**Comment by wsanchez** [comment]
Note: I used the wrong port number in the above build.

**Comment by wsanchez** [comment]
Fixed by #391
